FSubject: [PATCH] Move service from starting list once onStartCommand get called

Currently bg service will be marked as starting when it get started.
But it will be only get removed from the starting list when it get
binded. This caused the BOOT_COMPLETE intent delay more than 1min when
there are lot of service filter this intent.
Move service from the starting list if onStartCommand is completed.

diff --git a/services/core/java/com/android/server/am/ActiveServices.java b/services/core/java/com/android/server/am/ActiveServices.java
index 5ae77a7..b8f9bf9 100644
--- a/services/core/java/com/android/server/am/ActiveServices.java
+++ b/services/core/java/com/android/server/am/ActiveServices.java
@@ -2811,6 +2811,7 @@
             }
             final long origId = Binder.clearCallingIdentity();
             serviceDoneExecutingLocked(r, inDestroying, inDestroying);
+            getServiceMapLocked(r.userId).ensureNotStartingBackgroundLocked(r);
             Binder.restoreCallingIdentity(origId);
         } else {
             Slog.w(TAG, "Done executing unknown service from pid "
